ui/controls.js

```diff
@@ -522,6 +522,29 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     return this.seekBar_ ? this.seekBar_.getValue() : this.video_.currentTime;
   }
 
+  /**
+   * Checking the efficiency of a video
+   * @return {?number}
+   * @export
+   */
+  getBufferFillPercentage() {
+    const bufferedLength_ = this.video_.buffered.length;
+    const bufferedEnd_ =
+        bufferedLength_ ? this.video_.buffered.end(bufferedLength_ - 1) : 0;
+    const bufferingGoal_ = this.video_.currentTime +
+        Math.min(this.player_.getConfiguration().streaming.bufferingGoal,
+            (this.player_.seekRange().end - this.video_.currentTime));
+
+    if (bufferedEnd_ >= bufferingGoal_) {
+      return 100;
+    } else if (bufferedEnd_ < bufferingGoal_) {
+      return ((bufferedEnd_/bufferingGoal_)*100);
+    } else if (bufferedEnd_ <= this.video_.currentTime) {
```

Comment:
This condition needs to come before the previous one.  Otherwise, it will never be met.